spec/index.bs

```diff
@@ -6187,18 +6204,27 @@ attachments used by this encoder.
 
     : <dfn>writeTimestamp(querySet, queryIndex)</dfn>
     ::
+        Writes a timestamp value into |querySet| when all previous commands have completed executing.
 
         <div algorithm="GPURenderPassEncoder.writeTimestamp">
-            **Called on:** {{GPURenderPassEncoder}} this.
+            **Called on:** {{GPURenderPassEncoder}} |this|.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderPassEncoder/writeTimestamp(querySet, queryIndex)">
-                querySet:
-                queryIndex:
+                querySet: The query set that will store the timestamp values.
+                queryIndex: The index of the query in the query set.
             </pre>
 
             **Returns:** {{undefined}}
 
+            If any of the following conditions are unsatisfied, generate a validation error and stop.
+            <div class=validusage>
+                - |querySet| is [$valid to use with$] |this|.
+                - |querySet|.{{GPUQuerySet/[[type]]}} is {{GPUQueryType/timestamp}}.
+                - |queryIndex| &lt; |querySet|.{{GPUQuerySet/[[count]]}}.
+                - The query at same |queryIndex| cannot be written twice in the same |this|.
```

Comment:
is this forbidden in the native APIs?

Reply:
Yes, Vulkan does not allow to reset queries in render pass, but each query must be reset between uses, so we cannot write the same query twice in render pass.

Reply:
But why are you talking about resetting? I'd expect [vkCmdWriteTimestamp](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdWriteTimestamp.html) to be valid to call multiple times without a reset.

Reply:
We can call *vkCmdWriteTimestamp* multiple times, but if we call it with a same query index twice,  we need call [vkCmdResetQueryPool](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdResetQueryPool.html) to reset the query between these two *vkCmdWriteTimestamp*, otherwise it throws validation error:
```
vkCmdWriteTimestamp(..., 0 /*queryIndex*/);
vkCmdResetQueryPool(..., 0 /*firstQuery*/, 1 /*queryCount*/);
vkCmdWriteTimestamp(..., 0 /*queryIndex*/);
```

But the *vkCmdResetQueryPool* cannot be called in render pass, so we need this restriction.  On Vulkan 1.2, it provides a new API called [vkResetQueryPool](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/chap18.html#vkResetQueryPool) to reset query pool on the host, maybe we can remove this restriction in future.

Reply:
oh I see, it's VUID-vkCmdWriteTimestamp-None-00830